pkg/datapath/link/link.go

```diff
@@ -70,12 +65,34 @@ func GetIfIndex(ifName string) (uint32, error) {
 	return uint32(iface.Attrs().Index), nil
 }
 
-type ifNameCache struct {
+type LinkCache struct {
 	mu          lock.RWMutex
 	indexToName map[int]string
 }
 
-func (c *ifNameCache) syncCache() error {
+// NewLinkCache begins monitoring local interfaces for changes in order to
+// track local link information.
+func NewLinkCache() *LinkCache {
+	once.Do(func() {
+		linkCache = LinkCache{}
+
+		// Override the existing link-cache controller to update this copy
+		// we're currentyl constructing. If this function is called twice, then
```

Comment:
```suggestion
		// we're currently constructing. If this function is called twice, then
```

Reply:
:thumbsup: I don't think this comment was actually providing any value since I switched to using a sync.Once for initialization of the singleton, so I just dropped the whole comment.